### PR TITLE
Allow configuring different command bus backends

### PR DESCRIPTION
**Pull request by icewind1991**
This way apps have a safe way of providing a different command bus implementation.

also splits the shipped commandbus to allow for easier implementation of new busses.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: lib/private/Command/CronBus.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		if ($command instanceof \Closure) {
+			$serializer = new Serializer();
+			return $serializer->serialize($command);
+		} else if (is_callable($command) or $command instanceof ICommand) {
+			return serialize($command);
+		} else {
+			throw new \InvalidArgumentException('Invalid command');
+		}
+	}
+}
```
Comment (blizzz) [inline lib/private/Command/CronBus.php]: EOL missing

**Review by blizzz — APPROVED** [review]

File: tests/lib/Command/CronBusTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		return new CronBus($this->jobList);
+	}
+
+	protected function runJobs() {
+		$jobs = $this->jobList->getAll();
+		foreach ($jobs as $job) {
+			$job->execute($this->jobList);
+		}
+	}
+}
```
Comment (blizzz) [inline tests/lib/Command/CronBusTest.php]: here as well

File: lib/private/Command/CronBus.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/private/Command/CronBus.php]: Fixed.

File: tests/lib/Command/CronBusTest.php (hunk 2 above)
Comment (MorrisJobke) [reply tests/lib/Command/CronBusTest.php]: Fixed

**Pull request merged**